packages/timmy/timmy.1.0.0/opam

```diff
@@ -0,0 +1,50 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Time and calendar library"
+maintainer: ["mefyl <mefyl@gruntech.org>"]
+authors: ["mefyl <mefyl@gruntech.org>"]
+license: "LGPL-3.0-or-later"
+homepage: "https://github.com/mefyl/timmy"
+bug-reports: "https://github.com/mefyl/timmy/issues"
+depends: [
+  "dune" {>= "3.0"}
+  "base"
+  "fmt"
+  "ocaml" {>= "4.08.0"}
+  "ppx_here"
+  "ppx_deriving"
+  "ptime"
+  "alcotest" {with-test}
+  "odoc" {with-doc}
+  "cmdliner" {dev}
+  "opam-file-format" {dev}
+  "sexplib" {dev}
+  "stdio" {dev}
+  "shexp" {dev}
```

Comment:
What is the purpose of these dependencies?

Reply:
We have a utility executable (`.logistic/dune/extdeps/extdeps.exe`) that generates iOS cross compilation opam files amongst other things; it requires these. I've tried to get dune to NOT build it by default, but didn't succeed, thus I added them to be able to use OCaml CI. An alternative would be to erase these rules when building on OCaml-CI, but that would require maintaining a separate branch just for releases, which is a bit inconvenient.

Regarding dune, it seems there is no way to override the @check default alias, and `enabled_if` in `executable` stanzas support a very limited set of variables and notably not `lib_available:opam-file-format` :-/

Any tip appreciated !

Reply:
Ah i see :/ To workaround ocaml-ci's behaviour I can't see any cleaner solution than doing this:
```suggestion
  "cmdliner" {with-test & dev}
  "opam-file-format" {with-test & dev}
  "sexplib" {with-test & dev}
  "stdio" {with-test & dev}
  "shexp" {with-test & dev}
```
It's not the greatest but at least it works and does not bring unnecessary dependencies for regular users in 99% of cases.

Reply:
My understanding was that `dev` dependencies were not installed when installing via an archive (as opposed to a git repository), is it not the case ?

In any case, applied.

Reply:
yes, but we do support this use-case too. For example an user might want to do `opam pin --dev timmy` and in this case the dependencies shouldn't be needed.